### PR TITLE
Create s3-subnet-vpc-endpoint.py

### PR DESCRIPTION
**Pull request by pravinva**
AWS Proserve: AWS Config rules Mission 500 backlog

Verify VPC endpoint for S3 is configured in the route tables associated with the subnets

### CODE REVIEW

**Comment by pravinva** [comment]
closed a previous pull request to ensure squash didnt cause problems

File: python/subnet-s3-vpc-endpoint.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            "compliance_type": "NOT_APPLICABLE",
+            "annotation": "The rule doesn't apply to resources of type " +
+            configuration_item["resourceType"] + "."
+               }
+    subnet_id   = configuration_item["configuration"]["subnetId"]
+    vpc_id      = configuration_item["configuration"]["vpcId"]
+    client      = boto3.client("ec2");
+    s3vpcendpt  = False
+    route_tables = []
+    for resource_type in configuration_item["relationships"]:
```
Comment (gregjinkim) [inline python/subnet-s3-vpc-endpoint.py]: rename resource_type -> related_resource

File: python/subnet-s3-vpc-endpoint.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        if resource_type['resourceType']=='AWS::EC2::RouteTable':
+            route_table = resource_type['resourceId']
+ 
+        
+    vpcendpoints = client.describe_vpc_endpoints()
+    
+    for m in vpcendpoints['VpcEndpoints']:
```
Comment (gregjinkim) [inline python/subnet-s3-vpc-endpoint.py]: Can you rename m to be more descriptive?

File: python/subnet-s3-vpc-endpoint.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        if m['ServiceName'].endswith('.s3') and m['State'] == 'available' and m['VpcId']== vpc_id:
+            route_tables= m['RouteTableIds']
```
Comment (gregjinkim) [inline python/subnet-s3-vpc-endpoint.py]: I don't know what the route table check is for. Can you explain how this factors into the rule? And maybe include it in the description if you feel like that is important

**Comment by elsawyer** [comment]
@pravinva any word on this?

**Comment by jongogogo** [comment]
No response. Closing

**Pull request closed without merging**